addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="icloud" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<bridge-type id="account">
+		<label>iCloud Account</label>
+		<description>Provides various information for iCloud devices registered with this account.</description>
+
+		<config-description>
+			<parameter name="appleId" type="text" required="true">
+				<context>email</context>
+				<label>Apple Id</label>
+				<description>Apple Id (e-mail) to access iCloud information.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Apple iCloud password for the given Id.</description>
+			</parameter>
+			<parameter name="refreshTimeInMinutes" type="integer" min="5" max="65535" unit="min">
+				<label>Refresh</label>
+				<description>Refresh time in minutes.</description>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>iCloud Device</label>
+
+		<channels>
+			<channel typeId="BatteryStatus" id="batteryStatus" />
+			<channel typeId="system.battery-level" id="batteryLevel" />
+			<channel typeId="FindMyPhone" id="findMyPhone" />
+			<channel typeId="Location" id="location" />
+			<channel typeId="LocationAccuracy" id="locationAccuracy" />
+			<channel typeId="LocationLastUpdate" id="locationLastUpdate" />
+			<channel typeId="DeviceName" id="deviceName" />
+		</channels>
+
+		<representation-property>deviceId</representation-property>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device id</label>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+	
+	<channel-type id="Refresh">
+		<item-type>Switch</item-type>
+		<label>Refresh</label>
+		<description>Update all devices registered with this account immediately.</description>
+	</channel-type>
+	<channel-type id="BatteryStatus">
+		<item-type>String</item-type>
+		<label>Battery Status</label>
+	    <state>
+	        <options>
+	            <option value="NotCharging">Not charging</option>
+	            <option value="Charged">Charged</option>
+	            <option value="Charging">Charging</option>
+	            <option value="Unknown">Unknown</option>
+	        </options>
+	    </state>
+	</channel-type>
+	<channel-type id="FindMyPhone">
+		<item-type>Switch</item-type>
+		<label>Find my phone</label>
+	</channel-type>
+	<channel-type id="Location">
+		<item-type>Location</item-type>
+		<label>Location</label>

```

Comment:
add
```
		<state readOnly="true" />
```

Reply:
Also please consider to improve all labels. This is not very informative:

![grafik](https://user-images.githubusercontent.com/2870104/34003400-7ea6a28c-e0f5-11e7-9f4e-f97a9cbebc6c.png)

Reply:
No, I would stick to location - because in the UIs, you won't see "GPS Coordinates":

<img width="422" alt="screen shot 2017-12-14 at 17 45 39" src="https://user-images.githubusercontent.com/3244965/34003841-a708d140-e0f6-11e7-963f-9c9e2f39e331.png">

Reply:
Never used the UI part of Paper UI. Sounds pretty stupid. Okay let's not discuss this here any further.